applications/ShapeOptimizationApplication/python_scripts/mesh_controller_with_solver.py

```diff
@@ -76,6 +83,24 @@ def __init__(self, MeshSolverSettings, model):
         else:
             self.has_automatic_boundary_process = False
 
+        self.is_mesh_refinement_used = False
+        self.automatic_mesh_refinement_process_settings = self.MeshSolverSettings["automatic_mesh_refinement_settings"]
+        if (not self.automatic_mesh_refinement_process_settings.IsEquivalentTo(KM.Parameters("{}"))):
```

Comment:
To be in line with the setting above `"apply_mesh_solver" : true` i suggest to add a setting `use_automatic_remeshing: false` and use that flag to switch this behavior on/off.

Reply:
done :)